paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc

```diff
@@ -61,414 +61,425 @@ static mkldnn::memory::data_type GetDstType(bool is_int8, bool is_bfloat16,
+      PADDLE_ENFORCE_NE(bias->format(), MKLDNNMemoryFormat::undef,
                         platform::errors::InvalidArgument(
-                            "Wrong format set for Filter tensor"));
+                            "Got wrong format for Bias tensor."));
 
-      PADDLE_ENFORCE_GE(
-          input->dims().size(), 4,
-          platform::errors::InvalidArgument(
-              "Input must be with 4 or 5 dimensions, i.e. NCHW or "
-              "NCDHW, but got dimension = %d .",
-              input->dims().size()));
-      PADDLE_ENFORCE_LE(
-          input->dims().size(), 5,
-          platform::errors::InvalidArgument(
-              "Input must be with 4 or 5 dimensions, i.e. NCHW or "
-              "NCDHW, but got dimension = %d .",
-              input->dims().size()));
-
-      PADDLE_ENFORCE_GE(
-          filter->dims().size(), 4,
-          platform::errors::InvalidArgument(
-              "Filter must be with 4 or 5 dimensions, i.e. OIHW or "
-              "OIDHW, but got dimension = %d .",
-              filter->dims().size()));
-      PADDLE_ENFORCE_LE(
-          filter->dims().size(), 5,
-          platform::errors::InvalidArgument(
-              "Filter must be with 4 or 5 dimensions, i.e. OIHW or "
-              "OIDHW, but got dimension = %d .",
-              filter->dims().size()));
-
-      if (bias) {
-        PADDLE_ENFORCE_EQ(
-            bias->layout(), framework::DataLayout::kMKLDNN,
-            platform::errors::InvalidArgument(
-                "The Bias tensor's layout should be %d, but got %d.",
-                framework::DataLayout::kMKLDNN, bias->layout()));
-        PADDLE_ENFORCE_NE(bias->format(), MKLDNNMemoryFormat::undef,
-                          platform::errors::InvalidArgument(
-                              "Got wrong format for Bias tensor."));
-
-        PADDLE_ENFORCE_EQ(bias->dims().size(), 1,
-                          platform::errors::InvalidArgument(
-                              "Bias must only have 1 dimension, "
-                              "i.e. X, but got dimension = %d .",
-                              bias->dims().size()));
-      }
+      PADDLE_ENFORCE_EQ(
+          bias->dims().size(), 1,
+          platform::errors::InvalidArgument("Bias must only have 1 dimension, "
+                                            "i.e. X, but got dimension = %d .",
+                                            bias->dims().size()));
+    }
 
-      const std::string fuse_activation =
-          ctx.Attr<std::string>("fuse_activation");
-      const float fuse_alpha = ctx.Attr<float>("fuse_alpha");
-      const float fuse_beta = ctx.Attr<float>("fuse_beta");
-      const bool fuse_residual_conn =
-          ctx.Attr<bool>("fuse_residual_connection");
-      const int groups = ctx.Attr<int>("groups");
-      const std::string padding_algorithm =
-          ctx.Attr<std::string>("padding_algorithm");
+    const std::string fuse_activation =
```

Comment:
Will `const auto&` avoid copy here and following assignment ?

Reply:
This code I just touched due to indentation changes so I do not know. It sounds like a good idea to check that but I will not do that in this PR. Please add to our bugtracker.